File: obi/2021/p1/zero.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int n, num, total = 0;
    vector<int> numeros;
    cin >> n;

    while (n--) {
        cin >> num;
        if (num != 0) {
            numeros.push_back(num);
        } else {
            if (!numeros.empty()) {
                numeros.pop_back();
            }
        }
    }

    for (int i = 0; i < (int) numeros.size(); i++) {
        total += numeros[i];
    }

    cout << total << endl;
    return 0;
}

File: obi/2021/p1/zero.js
var n, numeros = [], total = 0;
scanf("%d", "n");

while (n--) {
   var num;
   scanf("%d", "num");
   if (num !== 0) {
        numeros.push(num);
   } else {
        if (numeros.length !== 0) {
            numeros.pop();
        }
   }
}

for (var i = 0; i < numeros.length; i++) {
    total += numeros[i];
}

printf("%d\n", total);